client/state/utils.ts

```diff
@@ -64,6 +61,13 @@ export function isValidStateWithSchema( state, schema, debugInfo ) {
 	return valid;
 }
 
+type CalypsoInitAction = Action< '@@calypso/INIT' >;
+
+interface KeyedState< S > {
+	[key: string]: S;
```

Comment:
This isn't actually the correct type because we can provide `_.get`-style paths as state keys, so the wrapped reducer's `S` state can be at an arbitrary path.

In practice, we never seem to do that so I'm wondering whether we should take a simple key and only operate on properties of the received state and not at some arbitrary property path.

Thoughts @Automattic/team-calypso?

Reply:
I agree: limiting the `keyPath` to just one property (no chaining) still covers 100% of usages and makes typing and reasoning much easier.

Reply:
I'd originally misinterpreted the way the `keyPath` works, I need to think about this a bit more 🤔